src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/Validation/IPluginValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Performance.SDK;
+using Microsoft.Performance.Toolkit.Plugins.Core.Metadata;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Runtime.Validation
+{
+    /// <summary>
+    ///     Validates that a plugin can be used.
+    /// </summary>
+    public interface IPluginValidator
+    {
+        /// <summary>
+        ///     Determines whether or not a plugin can be used.
+        /// </summary>
+        /// <param name="pluginMetadata">
+        ///     The metadata for the plugin attempting to be used.
+        /// </param>
+        /// <param name="errorInfos">
+        ///     <see cref="ErrorInfo"/>s describing why <paramref name="pluginMetadata"/> is invalid.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if <paramref name="pluginMetadata"/> is valid, <c>false</c> otherwise.
+        /// </returns>
+        bool IsValid(PluginMetadata pluginMetadata, out ErrorInfo[] errorInfos);
```

Comment:
>bool IsValid(PluginMetadata pluginMetadata, out ErrorInfo[] errorInfos); [](http://example.com/codeflow?start=8&length=72)

Do we ever use the 'bool' result? Should we just return ErrorInfo[] instead? (And rename the method if we do this.)

Reply:
I read the PluginsDiscoveryOrchestrator first and it wasn't using this return value, but FileBackedPluginsInstaller is. Still seems kinda weird. I still prefer to always just return ErrorInfo[] and then check for null/!any().